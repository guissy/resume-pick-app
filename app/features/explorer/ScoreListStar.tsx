import React from 'react';

import styles from './ScoreList.css';
import { calcLevelSalaryStar } from '../parser/tractWorkAge';

const Star: React.FC<{ style: React.CSSProperties }> = ({ style }) => {
  return (
    <span className={styles.star} role="img" aria-label="star" style={style}>
      ‚≠ê
    </span>
  );
};

type Props = {
  levelSalary: number;
  levelValue: number;
};
const ScoreListStar: React.FC<Props> = ({ levelSalary, levelValue }) => {
  const stars = calcLevelSalaryStar(levelSalary, levelValue);
  const arr = React.useMemo(() => {
    return stars > 0
      ? Array(Math.ceil(stars))
          .fill(0)
          .map((_v, i) => i)
      : [];
  }, [stars]);
  const lastStyle = React.useCallback(
    (i: number) => {
      const isLast = i === arr.length - 1;
      const isHalf = stars - Math.floor(stars) > 0.4;
      if (isLast && isHalf) {
        return { clipPath: 'inset(0 60% 0 0)' };
      }
      return {};
    },
    [arr, stars]
  );
  return (
    <span
      className={styles.namePhone}
      title={levelSalary.toFixed(2)}
      aria-label={stars.toFixed(1)}
    >
      {stars > 0
        ? arr.map((v, i) => <Star key={v} style={lastStyle(i)} />)
        : ''}
    </span>
  );
};

export default ScoreListStar;
