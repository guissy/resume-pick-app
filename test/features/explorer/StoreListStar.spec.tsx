import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ScoreListStar from '../../../app/features/explorer/ScoreListStar';

Enzyme.configure({ adapter: new Adapter() });

const defaultTestProps = {
  levelSalary: 1.12,
  levelValue: 8.48,
};

describe('ScoreListStar component', () => {
  it('should match exact snapshot', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-props-no-spreading
      .create(<ScoreListStar {...defaultTestProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match exact snapshot level 3', () => {
    const tree = renderer
      // eslint-disable-next-line react/jsx-props-no-spreading
      .create(<ScoreListStar {...defaultTestProps} levelSalary={1.33} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
