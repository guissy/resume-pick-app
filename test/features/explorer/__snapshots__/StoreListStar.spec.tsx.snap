// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreListStar component should match exact snapshot 1`] = `
<span
  aria-label="4.0"
  className="namePhone"
  title="1.12"
>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
</span>
`;

exports[`ScoreListStar component should match exact snapshot level 3 1`] = `
<span
  aria-label="4.5"
  className="namePhone"
  title="1.33"
>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={Object {}}
  >
    ⭐
  </span>
  <span
    aria-label="star"
    className="star"
    role="img"
    style={
      Object {
        "clipPath": "inset(0 60% 0 0)",
      }
    }
  >
    ⭐
  </span>
</span>
`;
